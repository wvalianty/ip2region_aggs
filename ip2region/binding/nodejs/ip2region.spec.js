'use strict';

const ip = require('./ip2region');

const query = ip.create('../../data/ip2region.db');

it('should query', () => {
  expect(query.btreeSearchSync('120.24.78.68')).toMatchSnapshot();
  expect(query.btreeSearchSync('10.10.10.10')).toMatchSnapshot();
});

it('binarySearch', () => {
  expect(query.binarySearchSync('120.24.78.68')).toMatchSnapshot();
  expect(query.binarySearchSync('10.10.10.10')).toMatchSnapshot();
});
