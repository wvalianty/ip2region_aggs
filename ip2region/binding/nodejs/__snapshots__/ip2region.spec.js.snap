// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`binarySearch 1`] = `
Object {
  "city": 2163,
  "region": "中国|华南|广东省|深圳市|阿里云",
}
`;

exports[`binarySearch 2`] = `
Object {
  "city": 0,
  "region": "未分配或者内网IP|0|0|0|0",
}
`;

exports[`should query 1`] = `
Object {
  "city": 2163,
  "region": "中国|华南|广东省|深圳市|阿里云",
}
`;

exports[`should query 2`] = `
Object {
  "city": 0,
  "region": "未分配或者内网IP|0|0|0|0",
}
`;
